Vic2ToHoI4/Source/HOI4World/Diplomacy/HoI4AIStrategy.cpp

```diff
@@ -0,0 +1,18 @@
+#include "HoI4AIStrategy.h"
+
+void HoI4::AIStrategy::updateStrategy()
+{
+	if (type == "befriend")
```

Comment:
I'd suggest using class enums and a switch whenever you have a branch like this, if only for an uninformed observer (like me) to have direct access to all valid types (and a default fallback).

Reply:
@Zemurin Well, all valid types are listed in [Vic2AI.cpp](https://github.com/ParadoxGameConverters/Vic2ToHoI4/pull/568/files#diff-b5d7ae4737cc3a8787efbcd585d63526)
`"conquer_prov|threat|antagonize|befriend|protect|rival"`
Since they are registered as std::string I can't use the switch on them. As for the default, the default is to not change the strategy type or any other part of the strategy.

Reply:
I see. Keep in mind it's perfectly legal to change stuff from other parts of the codebase to suit your needs and to improve them. If you wanted you could change that file over there to use enums, if that would make it easier to read as well.

Reply:
There are whole bunch of files that are parsed using std::string, I feel like it would only create inconsistency.

On another note, is it advisable to move this entire method into the HoI4AIStrategy constructor, thus converting the Vic2 strategy to corresponding HoI4 strategy at the creation?